lib/content-services/src/lib/search/components/search-logical-filter/search-logical-filter.component.spec.ts

```diff
@@ -0,0 +1,186 @@
+import { LogicalSearchCondition, LogicalSearchFields, SearchLogicalFilterComponent } from './search-logical-filter.component';
+
+describe('SearchLogicalFilterComponent', () => {
+    let component: SearchLogicalFilterComponent;
+    let fixture: ComponentFixture<SearchLogicalFilterComponent>;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            declarations: [SearchLogicalFilterComponent, SearchChipInputComponent],
+            imports: [
+                TranslateModule.forRoot(),
+                ContentTestingModule
+            ]
+        });
+
+        fixture = TestBed.createComponent(SearchLogicalFilterComponent);
+        component = fixture.componentInstance;
+        component.id = 'logic';
+        component.context = {
+            queryFragments: {
+                logic: ''
+            },
+            update: () => {}
+        } as any;
+        component.settings = { field: 'field1,field2', allowUpdateOnChange: true, hideDefaultAction: false };
+        fixture.detectChanges();
+    });
+
+    function getChipInputs(): HTMLInputElement[] {
+        return fixture.debugElement.queryAll(By.css('adf-search-chip-input input')).map((input) => input.nativeElement);
+    }
+
+    function getChipInputsLabels(): string[] {
+        return fixture.debugElement.queryAll(By.css('adf-search-chip-input mat-label')).map((label) => label.nativeElement.innerText);
+    }
+
+    function enterNewPhrase(value: string, index: number) {
+        const inputs = getChipInputs();
+        inputs[index].value = value;
+        fixture.detectChanges();
+        inputs[index].dispatchEvent(new KeyboardEvent('keydown', {keyCode: 13}));
+        fixture.detectChanges();
+    }
+
+    function clickApplyBtn() {
+        const applyBtn: HTMLButtonElement = fixture.debugElement.query(By.css('[data-automation-id="logical-filter-btn-apply"]')).nativeElement;
+        applyBtn.click();
+        fixture.detectChanges();
+    }
+
+    it('should update display value on init', () => {
+        spyOn(component.displayValue$, 'next');
+        component.ngOnInit();
+        expect(component.displayValue$.next).toHaveBeenCalledOnceWith('');
+    });
+
+    it('should not have valid value initially', () => {
+        expect(component.hasValidValue()).toBeFalse();
+    });
+
+    it('should contain 3 chip input components with correct labels', () => {
+        const labels = getChipInputsLabels();
+        expect(labels.length).toBe(3);
+        expect(labels[0]).toBe('SEARCH.LOGICAL_SEARCH.MATCH_ALL_LABEL');
+        expect(labels[1]).toBe('SEARCH.LOGICAL_SEARCH.MATCH_ANY_LABEL');
+        expect(labels[2]).toBe('SEARCH.LOGICAL_SEARCH.EXCLUDE_LABEL');
+    });
+
+    it('should has valid value after phrase is entered', () => {
+        enterNewPhrase('test', 0);
+        expect(component.hasValidValue()).toBeTrue();
+    });
+
+    it('should update display value after phrases changes', () => {
```

Comment:
This test looks like next one. Can we remove duplication?

Reply:
Removed